File: POOBanca/PHP/Revista.php
<?php
require_once "Produto.php";

class Revista extends Produto {
    protected $numero; // Integer

    // Override
    public function apresentacao() {
        parent::apresentacao();

        echo "O número da revista é {$this->numero}.<br/>";
    }

    // Construtor e Destrutor
    public function __construct($nome, $editora, $totPaginas, $numero) {
        parent::__construct($nome, $editora, $totPaginas);

        $this->numero = $numero;
    }

    public function __destruct() {
        echo "<p>Objeto Revista Destruído com Sucesso!</p>";
    }

    // Getters e Setters
    public function getNumero() {
        return $this->numero;
    }

    public function setNumero($numero) {
        $this->numero = $numero;
    }
}

File: POOBanca/Java/Revista.java
package poobanca;

public class Revista extends Produto {
    protected int numero;
    
    @Override
    public void apresentacao() {
        super.apresentacao();
        
        System.out.println("O número da revista é " + this.numero + ".");
    }
    
    // Construtor
    public Revista(String nome, String editora, int totPaginas, int numero) {
        super(nome, editora, totPaginas);
        
        this.numero = numero;
    }
    
    // Getters e Setters
    public int getNumero() {
        return this.numero;
    }
    
    public void setNumero(int numero) {
        this.numero = numero;
    }
}
